analyze/aps-cycle.c

```diff
@@ -319,6 +319,71 @@ bool instance_is_up(INSTANCE *i) {
   return (fibered_attr_direction(&i->fibered_attr)) == instance_outward;
 }
 
+static bool instance_is_local(INSTANCE *instance)
+{
+  if (instance->node ==  NULL) return true;
+
+  void* node = instance->node;
+  // Some instance->node are Expression instead of Declaration
+  if (ABSTRACT_APS_tnode_phylum(node) == KEYDeclaration)
+  {
+    switch (Declaration_KEY((Declaration)node))
+    {
+    // For black_dot(...).result to be considered local, the node of INSTANCE should be a proxy of function_decl
+    // Note that proxies are done using pragma_calls in aps-dnc
+    case KEYpragma_call: 
+    case KEYsome_function_decl:
+      return true;
```

Comment:
Hmm.  Maybe value_decl should also be here?  But we have to distinguish pattern variables ?x from a local declaration.